src/main/java/org/ow2/proactive/catalog/report/TableDataBuilder.java

```diff
@@ -0,0 +1,170 @@
+import org.springframework.stereotype.Component;
+
+import be.quodlibet.boxable.BaseTable;
+import be.quodlibet.boxable.Cell;
+import be.quodlibet.boxable.HorizontalAlignment;
+import be.quodlibet.boxable.Row;
+import be.quodlibet.boxable.image.Image;
+
+
+@Component
+public class TableDataBuilder {
+
+    private static final String LIGHT_GRAY = "#D3D3D3";
+    private static final String LIGHT_CYAN = "#F3F3F4";
+    private static final String WHITE = "#ffffff";
+    @Value("${pa.scheduler.url}")
+    private String schedulerUrl;
+
+    public void buildTableData(Set<CatalogObjectMetadata> orderedObjectsPerBucket, BaseTable table) {
+
+        // Create data header row
+        createDataHeader(table);
+
+        String currentBucketName = "";
+
+        for (CatalogObjectMetadata catalogObject : orderedObjectsPerBucket) {
+
+            if (!currentBucketName.equals(catalogObject.getBucketName())) {
+                currentBucketName = catalogObject.getBucketName();
+                Row<PDPage> dataRow = table.createRow(10f);
+                createDataCell(dataRow, (100), currentBucketName, 8);
+            }
+
+            Row<PDPage> dataRow = table.createRow(10f);
+            createDataCell(dataRow, (100 / 8f), catalogObject.getBucketName());
+            createDataCell(dataRow, (100 / 8f), getProjectName(catalogObject));
+            createDataCell(dataRow, (100 / 8f), catalogObject.getName());
+            createDataCell(dataRow, (100 / 8f) * 2, getDescription(catalogObject));
+            createDataCell(dataRow, (100 / 8f), catalogObject.getKind());
+            createDataCell(dataRow, (100 / 8f), catalogObject.getContentType());
+            createIconCell(dataRow, getIcon(catalogObject));
+
+        }
+        if (orderedObjectsPerBucket.isEmpty()) {
+            createEmptyTableRow(table);
+        }
+
+    }
+
+    private void createDataHeader(BaseTable table) {
+        Row<PDPage> factHeaderrow = table.createRow(15f);
+        createDataHeaderCell(factHeaderrow, (100 / 8f), "Bucket Name");
+        createDataHeaderCell(factHeaderrow, (100 / 8f), "Project Name");
+        createDataHeaderCell(factHeaderrow, (100 / 8f), "Object Name");
+        createDataHeaderCell(factHeaderrow, (100 / 8f) * 2, "Description");
+        createDataHeaderCell(factHeaderrow, (100 / 8f), "Kind");
+        createDataHeaderCell(factHeaderrow, (100 / 8f), "Content Type");
+        createDataHeaderCell(factHeaderrow, (100 / 8f), "Icon");
+        table.addHeaderRow(factHeaderrow);
+    }
+
+    private void createDataHeaderCell(Row<PDPage> factHeaderrow, float width, String title) {
+        Cell<PDPage> cell;
+        cell = factHeaderrow.createCell(width, title);
+        cell.setFontSize(8);
+        cell.setFillColor(java.awt.Color.decode(LIGHT_CYAN));
+        cell.setAlign(HorizontalAlignment.CENTER);
+    }
+
+    private void createDataCell(Row<PDPage> row, float width, String catalogObjectData, int size) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "size". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)